core/src/main/java/org/apache/accumulo/core/iteratorsImpl/system/MultiIterator.java

```diff
@@ -50,6 +51,7 @@ private MultiIterator(MultiIterator other, IteratorEnvironment env) {
     super(other.iters.size());
     this.iters = new ArrayList<>();
     this.fence = other.fence;
+    Collections.shuffle(other.iters);
```

Comment:
These shuffles will make seeks happen in different order.  There is still the case of initially opening the RFiles in FileManager that we could also shuffle.

Reply:
I believe I handled the FileManager case in bb6102a